src/compiler/checker.ts

```diff
@@ -27532,12 +27532,31 @@ namespace ts {
         function instantiateContextualType(contextualType: Type | undefined, node: Node, contextFlags: ContextFlags | undefined): Type | undefined {
             if (contextualType && maybeTypeOfKind(contextualType, TypeFlags.Instantiable)) {
                 const inferenceContext = getInferenceContext(node);
-                // If no inferences have been made, nothing is gained from instantiating as type parameters
-                // would just be replaced with their defaults similar to the apparent type.
-                if (inferenceContext && contextFlags! & ContextFlags.Signature && some(inferenceContext.inferences, hasInferenceCandidates)) {
-                    // For contextual signatures we incorporate all inferences made so far, e.g. from return
-                    // types as well as arguments to the left in a function call.
-                    return instantiateInstantiableTypes(contextualType, inferenceContext.nonFixingMapper);
+                if (inferenceContext && contextFlags! & ContextFlags.Signature) {
+                    // If no inferences have been made, nothing is gained from instantiating as type parameters
+                    // would just be replaced with their defaults similar to the apparent type.
+                    if (some(inferenceContext.inferences, hasInferenceCandidates)) {
+                        // For contextual signatures we incorporate all inferences made so far, e.g. from return
+                        // types as well as arguments to the left in a function call.
+                        return instantiateInstantiableTypes(contextualType, inferenceContext.nonFixingMapper);
+                    }
+
+                    const apparentType = mapType(contextualType, getApparentType, /*noReductions*/ true);
+
+                    // If the apparent type is any, we cannot lose any information by instantiating,
+                    // so we might as well try and possibly get a useful type
+                    if (apparentType === anyType) {
+                        return instantiateInstantiableTypes(contextualType, inferenceContext.nonFixingMapper);
+                    }

```

Comment:
In what cases this is helpful? It's not totally clear to me what exactly the apparent type is but if I take into account the comment above the `getApparentType` then it should return the base constraint for type parameters.

Given that this PR introduces only tests that have a base constraint that is not `any` - I wonder, what existing test cases does this branch satisfy?